cypress/e2e/models/migration/applicationinventory/assessment.ts

```diff
@@ -213,18 +212,27 @@ export class Assessment extends Application {
             commonView.alertTitle,
             `Success alert:Success! Assessment discarded for ${this.name}.`
         );
-        this.perform_assessment(risk, stakeholders, stakeholderGroups, (applicationOpen = false));
+        this.perform_assessment(risk, stakeholders, stakeholderGroups, false);
     }
 
-    take_questionnaire(): void {
-        clickByText(button, "Take");
+    take_questionnaire(questionnaireName?: string): void {
+        if (!questionnaireName) {
+            clickByText(button, "Take");
+        } else {
+            // Wait for the element to load
+            cy.contains(questionnaireName).should("be.visible");
+
+            // click on the take button for that questionair
+            cy.contains(questionnaireName).siblings("td").contains("button", "Take").click();
```

Comment:
This can be achieved in a single statement.

                cy.contains(questionnaireName).should("be.visible").siblings("td").contains("button", "Take").click();;